docs/developer/extending/apps/synchronous-webhooks/transaction-webhooks.mdx

```diff
@@ -0,0 +1,432 @@
+---
+title: Transaction webhooks
+---
+
+# Introduction
+
+:::caution
+This feature is currently in **Feature Preview**. This part of Saleor is not complete
+and subject to change but is available to experiment and provide feedback.
+:::
+
+Transaction webhooks are synchronous webhooks that allow delegating the action for transaction to Saleor App.
+Synchronous means that these webhooks expect a response of a particular
+shape to be returned from the App to continue processing in Saleor. Transaction webhooks only
+support the HTTP(S) protocol; they are sent as POST requests with the application/json body
+and expect a response of the same content type.
+
+## Key concepts
+
+The `HANDLE_PAYMENTS` permission is required for the App to be able to receive transaction webhooks.
+The webhook notification is only sent to a specific app that is connected to given transaction.
+
+## General transaction flow
+
+The usage of the webhook is strictly related to the payment flow handled by transactions.
+More details about whole transaction flow can be found
+[here](../../../payments#saleor-app-recommended).
+
+## Transaction charge
+
+`TRANSACTION_CHARGE_REQUESTED`
+
+:::info
+
+This feature was introduced in **Saleor 3.12**.
+
+:::
+
+Synchronous webhook called inside celery task. Called when staff user requests the charge action.
+It expects in the response at least `pspReference`. The `pspReference` will be attached to [TransactionEvent](api-reference/objects/transaction-event.mdx)
+with request details. Attached `pspReference` to "request" object means that the webhook has been
+successfully processed by app. Optionally the result data can be provided. The data is
+used to provide the final status of the requested action, and it will be used to create a new [TransactionEvent](api-reference/objects/transaction-event.mdx)
+object. The new [TransactionEvent](../../../../api-reference/objects/transaction-event.mdx) object will have provided `pspReference`, and will be used in the
+[re-calculation of the transaction's amount process](../../../payments#re-calculations-of-transaction-amounts).
+
+### Request
+
+Saleor will send a
+[TRANSACTION_CHARGE_REQUESTED](../../../../api-reference/enums/webhook-event-type-sync-enum#code-style-fontweight-normal-webhookeventtypesyncenumbtransaction_charge_requestedbcode)
+webhook by using [TransactionChargeRequested](../../../../api-reference/objects/transaction-charge-requested) subscription type
+or with pre-defined payload in case of missing subscription query.
```

Comment:
```suggestion
webhook by using [TransactionChargeRequested](../../../../api-reference/objects/transaction-charge-requested) subscription type or with a pre-defined payload in case of a missing subscription query.
```